ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/streaming/WSQueryEndpoint.java

```diff
@@ -279,16 +272,28 @@ private KsqlRequest parseRequest(final MultiMap requestParams) {
     }
   }
 
+  private void attachCloseHandler(final ServerWebSocket websocket,
+                                  final WebSocketSubscriber<?> subscriber) {
+    websocket.closeHandler(v -> {
+      subscriber.close();
```

Comment:
can subscriber be null? I see that validation in the code you removed. Should it be checked before creating the closeHandler?

Reply:
There isn't really a path that makes it null, but it makes sense to be more defensive there, I'll bring back the check!